src/builder.rs

```diff
@@ -102,6 +117,7 @@ impl<'ctx> Builder<'ctx> {
     /// builder.build_return(Some(&i32_arg));
     /// ```
     pub fn build_return(&self, value: Option<&dyn BasicValue<'ctx>>) -> InstructionValue<'ctx> {
+        check_positioned!(self);
```

Comment:
Rather than asserting, I think we should return a Result to indicate an error has occurred

Reply:
It'd be good to start using the `thiserror` crate. Could add a `BuilderError` type

Reply:
I agree - that would make the check more visible. However, it would be an API change, and users may need to update large amounts of their code to either handle an `Err` or extract the `Ok` value.

Of course, maybe this is a good thing, but how often do programmers make the mistake of not setting a position?

Reply:
Being a pre-1.0 crate, we aren't making any API stability guarantees. We should go ahead and do this. If people want the old functionality, unwrapping will achieve the same result